### PR TITLE
Update minimum versions of scipy and mypy

### LINKED ISSUE #126: Installing dependencies fails
Tested on fresh Ubuntu 22.04 install, python 3.10.4, pip 22.0.4, setuptools 62.1.0.

`scipy==1.5.3` and `typed-ast==1.4.3` fail to build, there are no wheels available.

**Reproduce:**

`pip install scipy==1.5.3`

**Possible Solutions**

- Increase minimum versions of these packages.
- Add necessary packages/libraries (python-dev, LAPACK/BLAS) to build dependencies to install guide

### PR DESCRIPTION
**Pull request by k-collie**
Fixes #126 

Bump scipy and mypy versions that were causing problems, move and fix mypy config.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,11 +40,11 @@ readme = 'README.md'  # Markdown files are supported
 include = [".git", "indica/git_version"]
 
 [tool.poetry.dependencies]
-python = "^3.8"
-prov = "^1.5.3"
+python = ">=3.8,<3.11"
```
Comment (k-collie) [inline pyproject.toml:43]: `scipy` started adding maximum python versions in 1.6.2, so `poetry` now complains if we use the `^` here

Comment (k-collie) [reply pyproject.toml]: This is an unnecessary restriction of poetry that I don't think `pip` has, every dependency version must be compatible with every python version.

File: pyproject.toml [hunk 2]
```diff
@@ -40,11 +40,11 @@ readme = 'README.md'  # Markdown files are supported
 include = [".git", "indica/git_version"]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.8,<3.11"
 prov = "^1.5.3"
 netCDF4 = "^1.5.4"
 numpy = "^1.21.0"
-scipy = "^1.5.2"
+scipy = "^1.6.2"
```
Comment (k-collie) [inline pyproject.toml:47]: 1.6.2 is the first release which specifies a maximum python version

**Review by elitherl — APPROVED** [review]
Update looks good

**Pull request merged**